src/components/Calculator.js

```diff
@@ -1,55 +1,72 @@
 /* eslint-disable react/prefer-stateless-function */
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
```

Comment:
- [ ] Please validate the props, Properties validation is a useful way to force the correct usage of the components. This will help during development to avoid future bugs and problems, once the app becomes larger. It also makes the code more readable, since we can see how each component should be used